lib/analysis/network.py

```diff
@@ -375,20 +373,26 @@ def channel_user_presence_graph_and_csv(nicks, nick_same_list, channels_for_user
     return presence_graph_and_matrix, full_presence_graph
 
 
-def degree_analysis_on_graph(nx_graph):
+def degree_analysis_on_graph(nx_graph, date=None):
     """
         perform degree analysis of input graph object
-
     Arguments:
         nx_graph (nx_object): object to perform analysis on
-
     Returns:
         null
     """
-
-    nodes_with_OUT_degree = [["nodes_w_out_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.out_degree().values())+1))]
-    nodes_with_IN_degree = [["nodes_w_in_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.in_degree().values())+1))]
-    nodes_with_TOTAL_degree = [["nodes_w_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.degree().values())+1))]
+    if len(nx_graph.out_degree().values()):
+        nodes_with_OUT_degree = [["nodes_w_out_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.out_degree().values())+1))]
+    else:
+        nodes_with_OUT_degree = [["NA", 0, "NA"]]
+    if len(nx_graph.in_degree().values()):
+        nodes_with_IN_degree = [["nodes_w_in_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.in_degree().values())+1))]
+    else:
+        nodes_with_IN_degree = [["NA", 0, "NA"]]
+    if len(nx_graph.degree().values()):
+        nodes_with_TOTAL_degree = [["nodes_w_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.degree().values())+1))]
+    else:
+        nodes_with_TOTAL_degree = [["NA", 0, "NA"]]
 
     if config.DEBUGGER:
         print nx_graph.out_degree()
```

Comment:
this was introduced by you. but i believe this is being done to check if doesn't return an empty result other it may create an error with this nodes_with_TOTAL_degree = [["nodes_w_deg" + str(i), 0, ''] for i in xrange((max(nx_graph.degree().values())+1))]